File: core/src/main/java/vn/loitp/core/utilities/LFCMUtil.java
package vn.loitp.core.utilities;

/**
 * Created by LENOVO on 6/7/2018.
 */

import android.os.AsyncTask;

import org.json.JSONObject;

import loitp.core.R;
import okhttp3.MediaType;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.RequestBody;
import okhttp3.Response;
import vn.loitp.core.common.Constants;

/**
 * Created by HOME on 4/18/2017.
 */

public class LFCMUtil {
    private static final MediaType JSON = MediaType.parse("application/json; charset=utf-8");

    public static void sendNotification(final String key, final String body) {
        new AsyncTask<Void, Void, Void>() {
            @Override
            protected Void doInBackground(Void... params) {
                try {
                    OkHttpClient client = new OkHttpClient();
                    JSONObject json = new JSONObject();
                    JSONObject dataJson = new JSONObject();
                    dataJson.put("body", body);
                    dataJson.put("title", R.string.app_name);
                    json.put("notification", dataJson);
                    json.put("to", Constants.INSTANCE.getFCM_TOPIC());
                    RequestBody body = RequestBody.create(JSON, json.toString());
                    //LLog.d(TAG, "body:" + LApplication.getGson().toJson(body));
                    Request request = new Request.Builder()
                            .header("Authorization", "key=" + key)
                            .url("https://fcm.googleapis.com/fcm/send")
                            .post(body)
                            .build();
                    Response response = client.newCall(request).execute();
                    String finalResponse = response.body().string();
                    //LLog.d(TAG, "finalResponse:" + LApplication.getGson().toJson(finalResponse));
                } catch (Exception e) {
                    //LLog.d(TAG, "Exception " + e.toString());
                }
                return null;
            }
        }.execute();
    }
}

File: core/src/main/java/com/core/utilities/LFCMUtil.kt
package com.core.utilities

import android.annotation.SuppressLint
import com.R
import com.core.common.Constants
import io.reactivex.Completable
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import okhttp3.MediaType.Companion.toMediaTypeOrNull
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.RequestBody.Companion.toRequestBody
import org.json.JSONObject

/**
 * Created by Loitp on 4/18/2017.
 */

class LFCMUtil {
    companion object {
        private val TAG = LFCMUtil::class.java.simpleName
        private val JSON = "application/json; charset=utf-8".toMediaTypeOrNull()

        @SuppressLint("CheckResult")
        fun sendNotification(key: String, body: String) {
            Completable.fromAction {
                val client = OkHttpClient()
                val json = JSONObject()
                val dataJson = JSONObject()
                dataJson.put("body", body)
                dataJson.put("title", R.string.app_name)
                json.put("notification", dataJson)
                json.put("to", Constants.FCM_TOPIC)
                val jsonBody = json.toString().toRequestBody(JSON)
                val request = Request.Builder()
                        .header("Authorization", "key=$key")
                        .url("https://fcm.googleapis.com/fcm/send")
                        .post(jsonBody)
                        .build()
                val response = client.newCall(request).execute()
                val finalResponse = response.body
                LLog.d(TAG, "finalResponse:" + finalResponse.toString())
            }.subscribeOn(Schedulers.io())
                    .observeOn(AndroidSchedulers.mainThread())
                    .subscribe({
                        LLog.d(TAG, "onComplete")
                    }, {
                        LLog.d(TAG, "onError $it")
                    })
        }

//        fun getFCMToken(context: Context) {
//            FirebaseInstanceId.getInstance().instanceId
//                    .addOnCompleteListener(OnCompleteListener { task ->
//                        if (!task.isSuccessful) {
//                            return@OnCompleteListener
//                        }
//                        val fcmToken = task.result?.token
//                        VinPref.setFCMToken(fcmToken)
//                    })
//        }

//        fun resetInstanceId() {
//            Thread(Runnable {
//                try {
//                    FirebaseInstanceId.getInstance().deleteInstanceId()
//                    FirebaseInstanceId.getInstance().instanceId
//                } catch (e: IOException) {
//                    e.printStackTrace()
//                }
//            }).start()
//        }
    }
}
